.github/workflows/sns_lifecycle.yml

```diff
@@ -0,0 +1,67 @@
+name: "SNS lifecycle"
+
+on:
+  push:
+     branches: [ main ]
+  pull_request:
+
+jobs:
+  sns_lifecycle:
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Build docker image
+      run: |
+        docker build -t dfinity/sns-testing .
+    - name: Start docker image
+      run: |
+        SNS_TESTING_INSTANCE=$(
+          docker run -p 8080:8080 -v "`pwd`":/dapp -d dfinity/sns-testing dfx start --clean
+        )
+        while ! docker logs $SNS_TESTING_INSTANCE 2>&1 | grep -m 1 'Dashboard:'
+        do
+          echo "Awaiting local replica ..."
+          sleep 3
+        done
+    - name: Set up docker image
+      run: |
+        SNS_TESTING_INSTANCE=$(docker ps -q)
+        docker exec $SNS_TESTING_INSTANCE bash setup_locally.sh
+    - name: Run basic scenario
+      run: |
+        SNS_TESTING_INSTANCE=$(docker ps -q)
+        docker exec $SNS_TESTING_INSTANCE bash run_basic_scenario.sh
+    - name: NNS frontend dapp smoke test
+      run: |
+        curl http://qsgjb-riaaa-aaaaa-aaaga-cai.localhost:8080/launchpad/ | gzip -d | grep "<title>Network Nervous System frontend dapp</title>"
+    - name: Stop docker container
+      run: |
+        SNS_TESTING_INSTANCE=$(docker ps -q)
+        docker kill $SNS_TESTING_INSTANCE
+
+  publish:
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+    needs: sns_lifecycle
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Log in to Docker Hub
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
Who will effectively administer these secrets? IDX?

Reply:
I've reached out to IT about that.

Reply:
Factored out from this PR.